### DISCUSSION TITLE
How can I execute a JS function on user signup?

### DISCUSSION THREAD

**Discussion by 16Integer** [supabase/supabase#20613] [Questions] [answered]
I would like to execute a JS function on user signup to perform some actions, such as inserting data into the profiles table. I know I can do this through the SQL editor and a trigger, however, as I have API requests that need to be made with environment variables found in the Node.js environment, and http requests need to be made, for simplicity, I'd prefer to handle this on the JS client.

How can I achieve this?

**Comment by GaryAustin1** [comment, marked as answer]
Can you do them when the callback comes to your app from the user confirming their email or OAuth info?
You can control the url that is returned to, and then update your profiles table as needed server side.

You can use a Postgres trigger function with the pg_net or http extension to call pretty much any endpoint if you can create one.
